src/box/lua/upgrade.lua

```diff
@@ -314,9 +314,33 @@ space_formats_1_7_5._vuser = space_formats_1_7_5._user
 space_formats_1_7_5._vpriv = space_formats_1_7_5._priv
 
 local function space_trig_1_7_5(_, tuple)
-    if tuple and space_formats_1_7_5[tuple[3]] and
+    if tuple == nil then
+        return nil
+    end
+    local is_changed = false
+    if space_formats_1_7_5[tuple[3]] and
        not table.equals(space_formats_1_7_5[tuple[3]], tuple[7]) then
+        -- Update format of system spaces.
         tuple = tuple:update{{'=', 7, space_formats_1_7_5[tuple[3]]}}
+        is_changed = true
+    elseif type(tuple[7]) == 'table' and next(tuple[7]) ~= nil and
+           type(tuple[7][1]) == 'table' and tuple[7][1]['name'] == nil then
+        -- Update format of user spaces. It had the following type:
+        -- {'a', type = 'number'}, and not it looks like that:
+        -- {name = 'a', type = 'number'}.
+        local format = tuple[7]
+        for _, f in ipairs(format) do
+            f['name'] = f[1]
+            f[1] = nil
+            is_changed = true
+        end
+        tuple = tuple:update{{'=', 7, format}}
+    elseif type(tuple[7]) ~= 'table' or not next(tuple[7]) then
```

Comment:
I am wondering why here it is `not next(tuple[...])` and in the seemingly same places above it is `next(tuple[...]) == / != nil`? If this can be `box.NULL`, then shouldn't we always explicitly compare with `nil`?

Reply:
I suppose, it was just added on different iterations of review and this is the result.  Let's explicitly use comparison with `nil` everywhere, since we don't wanna see `{box.NULL}` arrays in the format anyway.